clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -222,4 +245,108 @@ private KeyStore load() throws GeneralSecurityException, IOException {
         }
     }
 
+
+    private class ReloadableX509TrustManager extends X509ExtendedTrustManager implements X509TrustManager {
+        private final SecurityStore trustStore;
+        private TrustManagerFactory tmf;
+        private X509TrustManager trustManager;
+        private long lastReload = 0L;
+        private long minimumDelay = 60 * 1000L;
+
+        public ReloadableX509TrustManager(SecurityStore trustStore, TrustManagerFactory tmf) {
+            this.trustStore = trustStore;
+            this.tmf = tmf;
+        }
+
+        public ReloadableX509TrustManager(SecurityStore trustStore, TrustManagerFactory tmf, long minimumDelay) {
```

Comment:
This is in a private class, so tests won't be able to configure the delay? It will be good to add some unit tests and end-to-end tests as well if possible.